Lib/glyphsLib/builder/components.py

```diff
@@ -51,6 +69,50 @@ def to_ufo_components(self, ufo_glyph, layer):
             ufo_glyph.lib[_lib_key(key)] = values
 
 
+def to_ufo_components_decompose(self, ufo_glyph, layer):
+    """Draw decomposed .glyphs components with a pen, adding them to the parent
+    glyph."""
+
+    layer_id = layer.foreground.layerId
```

Comment:
there is no mention in the method name or docstring that this applies only to background layers, but here you are accessing a property which is only present on GSBackgroundLayer objects. Would make sense to rename `to_ufo_background_components_decompose` or similar